null_safety_examples/cookbook/networking/background_parsing/lib/main_step3.dart

```diff
@@ -1,42 +1,39 @@
 import 'dart:async';
 import 'dart:convert';
-
+import 'package:flutter/foundation.dart';
 import 'package:http/http.dart' as http;
 
 // #docregion parsePhotos
 // A function that converts a response body into a List<Photo>.
 List<Photo> parsePhotos(String responseBody) {
-  final parsed = jsonDecode(responseBody).cast<Map<String, dynamic>>();
-
-  return parsed.map<Photo>((json) => Photo.fromJson(json)).toList();
+  final parsed = jsonDecode(responseBody) as List;
+  return [for (final json in parsed) Photo.fromJson(json)];
 }
 
 Future<List<Photo>> fetchPhotos(http.Client client) async {
-  final response = await client
-      .get(Uri.parse('https://jsonplaceholder.typicode.com/photos'));
+  final uri = Uri.parse('https://jsonplaceholder.typicode.com/photos');
+  final response = await client.get(uri);
 
   // Use the compute function to run parsePhotos in a separate isolate.
-  return parsePhotos(response.body);
+  return compute(parsePhotos, response.body);
```

Comment:
In the step 3, we don't use the `compute` function yet, the change is made in the step 4.

The comment in the line 17 should be removed instead.